src/models/basemodel.py

```diff
@@ -85,13 +89,13 @@ def check_data(data, fields, required_fields=[]):
         for field in required_fields:
             if field not in data:
                 yield (field, 'Missing field: {}'.format(field))
```

Comment:
keep it. see: http://stackoverflow.com/questions/231767/what-does-the-yield-keyword-do-in-python